code_challenges/code_wars/replace_with_alphabet_position.py

```diff
@@ -0,0 +1,10 @@
+# https://www.codewars.com/kata/546f922b54af40e1e90001da/train/python
+
+def replace_characters(characters):
+    characters = characters.lower()
+    position_index = ["a","b","c","d","e","f","g","h","i","j","k","l","m","n","o","p","q","r","s","t","u","v","w","x","y","z"]
+    positions = ""
+    for character in characters:
+        if character.isalpha():
+            positions += f"{position_index.index(character) + 1} "
+    return positions.rstrip()
```

Comment:
I don't readily see how the spaces are being added to positions? 

The `.index(character)` method isn't very efficient, it has to look through the whole list to find the index.  Since it has to be written out anyway, you might use a dict like `{"a": 1, "b": 2...}`.  The lookup would then be o(1) instead of needing to search the whole alphabet each time.

There is probably also a way to use the [built in ord](https://docs.python.org/3/library/functions.html#ord) method on the lower letter and some math to get the number.

Reply:
Oh I see `f"{position_index.index(character) + 1} "` has a space at the end that is then trimmed.